app/src/org/commcare/dalvik/activities/EntityMapActivity.java

```diff
@@ -1,288 +1,186 @@
-                                }
-                            } catch (StorageFullException | IOException e1) {
-                                e1.printStackTrace();
-                            }
-                        }
-                        
-                        //Ok, so now we have an address or not. If we _do_ have one, let's have some fun
-                        
-                        if(gp != null) {
-                            OverlayItem overlayItem = overlayFactory.generateOverlay(gp, e);
-                            mEntityOverlay.addOverlay(overlayItem, e.getElement());
-                        }
-                        else { 
-                            bogus++;
+                    String address = entity.getFieldString(i).trim();
+                    if(address != null && !"".equals(address)) {
+                        LatLng location = getLatLngFromAddress(address);
+                        if (location == null) {
+                            bogusAddresses++;
+                        } else {
+                            entityLocations.add(new Pair<>(entity, location));
                         }
                     }
                 }
             }
         }
-        
-        Log.d(TAG, "Loaded. " + legit +" addresses discovered, " + bogus + " could not be located");
-
-        if (legit != 0 && mEntityOverlay.getCenter() != null) {
-            map.getController().animateTo(mEntityOverlay.getCenter());
-        } else if(location != null) {
-            int lat = (int) (location.getLatitude() * 1E6);
-            int lng = (int) (location.getLongitude() * 1E6);
-            GeoPoint point = new GeoPoint(lat, lng);
-            map.getController().animateTo(point);
-        }
-        
-        map.getOverlays().add(mMyLocationOverlay);
-        //The overlay crashes out if you try to draw it and it's empty,
-        //so only add it if we found something.
-        if(mEntityOverlay.size() > 0) {
-            map.getOverlays().add(mEntityOverlay);
-        }
-        map.getController().setZoom(18);
-        map.setClickable(true);
-        map.setEnabled(true);
-        Log.d(TAG, "Done loading");
+        Log.d(TAG, "Loaded. " + entityLocations.size() +" addresses discovered, " + bogusAddresses + " could not be located");
     }
-    
-    private String getLicenseKey() {
-        //If there's a defined debug key in the local environment, use that.
-        int debugId = this.getResources().getIdentifier("maps_api_key_debug","string", this.getPackageName());
-        if(debugId == 0) { 
-            return BuildConfig.MAPS_API_KEY;
+
+    private LatLng getLatLngFromAddress(String address) {
+        LatLng location = null;
+        try {
+            GeoPointData data = new GeoPointData().cast(new UncastData(address));
+            if(data != null) {
+                location = new LatLng(data.getLatitude(), data.getLongitude());
+            }
+        } catch(Exception ex) {

```

Comment:
Yeah, been meaning to start using them more myself. They are only at the android level, so don't think they work with vanilla java, such as our javarosa/commcare repos.